skimage/transform/_warps.py

```diff
@@ -633,11 +633,13 @@ def warp_coords(coord_map, shape, dtype=np.float64):
     return coords
 
 
-def _clip_warp_output(input_image, output_image, order, mode, cval, clip):
+def _clip_warp_output(input_image, output_image, order, mode, cval, clip,
```

Comment:
I don't see why `_clip_warp_output` needs to know anything about anti-aliasing.

Reply:
This is because `_clip_warp_output` doesn't perform clipping when `order == 0` since values are not suppose to go outside input values range. But this is no more true when `anti_aliasing` is activated :confused:. @abouysso pointed this here https://github.com/scikit-image/scikit-image/pull/5209#issuecomment-769958890

Reply:
I'm missing something: the idea is that we don't *need* to clip because of order 0.  But if out of range data makes it way **in**, that should probably be an error, unless preserve range or similar is set (otherwise, what's the point of clipping).

Reply:
Exact! However out of range data can makes it way in because of `ndi.gaussian_filter` called when anti_aliasing is True. This can be considered as a bug from scipy package. Here the proposition is to use `_clip_warp_output` to handle out of range data caused by interpolation and `ndi.gaussian_filter` !

Reply:
The case we are addressing, though, is surely one that the user did not anticipate.  If they send in out-of-range values, they likely did *not* intend for values to be clipped.

Reply:
Yes, but this behavior is controlled by the `clip`and `preserve_range` parameters... I mean, in such case, the user have to set `preserve_range` to `True` and `clip` to `False`...

Reply:
The problem could occur like this, as an example:
- user sends in data in range [0,1] for downsizing, and sets `anti_aliasing=True`, `order=0` and `clip=True`, expecting output to be also in range [0,1]
- data is first filtered with `ndi.gaussian_filter`, which can produce out of range data
- the actual resize operation is performed
- `_clip_warp_output` is called and does nothing since `order=0`
- the user get output with out of range data caused by the first step, while `clip=True`... My opinion is that is unexpected behavior.

That's why I propose that `_clip_warp_output` knows about `anti_aliasing`, to clip out of range data potentially caused by anti_aliasing operation ! :smiley:

Reply:
Why not simply always clip inside of `_clip_warp_output`, and call it only when `clip=True`?

Reply:
That's an other option, but avoiding clipping in the special case `order == 0 or anti_aliasing` saves traversing the image twice searching for its min and max values...

Reply:
Given that the standard code path does it anyway, I'd opt for simplicity of code.

Reply:
Previous implementation do not clip values when `order == 0`. If I got you well, you prefer to also remove this option, right?

Reply:
I think we should allow clipping whenever overflow is a possibility; so, since for order 0 overflow cannot occur, it is fine to skip the clipping entirely.

Reply:
But overflow can occur when order == 0 and anti_aliasing is True like in the situation described by @abouysso https://github.com/scikit-image/scikit-image/pull/5209#discussion_r568468260

Reply:
That's the root of the reported bug :)

Reply:
To illustrate this:
```python
import numpy as np
from skimage.transform import resize
img = np.ones((5, 5))
resized = resize(img, (3, 3), order=0, anti_aliasing=True, clip=True)
resized.max()  # output: 1.0000000000000004
```

Reply:
(if `clip=True`)

Reply:
Alright, thank you for your input !